python/ray/train/lightning/_lightning_utils.py

```diff
@@ -102,9 +102,7 @@ class RayDeepSpeedStrategy(DeepSpeedStrategy):
     """Subclass of DeepSpeedStrategy to ensure compatibility with Ray orchestration."""
 
     def setup_distributed(self):
-        # Device ranks have already been specified in RayEnvironment
-        # Clear parallel_devices to skip deepspeed local rank checks
-        self.parallel_devices = []
+        self.parallel_devices = list(range(torch.cuda.device_count()))
```

Comment:
Would something like this make sense? Not sure if the devices indices are always starting from 0.

```suggestion
        devices = train.torch.get_device()
        if not isinstance(devices, list):
            devices = [devices]
        self.parallel_devices = [d.index for d in devices if d.index is not None]
```

Reply:
The `parallel_devices` here need to have all the cuda device ids for current node. `train.torch.get_device()` only return the one for the current worker.

For example, if we have two workers each with 1 GPU
- worker_0: `CUDA_VISIBLE_DEVICES=2,3`, `train.torch.get_device()=torch.device(cuda:0)`
- worker_1: `CUDA_VISIBLE_DEVICES=2,3`, `train.torch.get_device()=torch.device(cuda:1)`

`parallel_devices` here should be `[torch.device(cuda:0), torch.device(cuda:1)]`